sunpy/coordinates/utilities.py

```diff
@@ -0,0 +1,208 @@
+    earth = SkyCoord(earth_icrs, frame='icrs', obstime=obstime).transform_to(HGS)
+
+    # Explicitly set the longitude to 0
+    earth = SkyCoord(0*u.deg, earth.lat, earth.radius, frame=earth)
+
+    return earth
+
+
+def get_sun_B0(time='now'):
+    """
+    Return the B0 angle for the Sun at a specified time, which is the heliographic latitude of the
+    Sun-disk center as seen from Earth.  The range of B0 is +/-7.23 degrees.
+
+    Parameters
+    ----------
+    time : various
+        Time to use as `~astropy.time.Time` or in a parse_time-compatible format
+
+    Returns
+    -------
+    out : `~astropy.coordinates.Angle`
+        The position angle
+    """
+    return Angle(get_earth(time).lat)
+
+
+# Ignore warnings that result from going back in time to the first Carrington rotation
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore", ErfaWarning)
+
+    # Carrington rotation 1 starts late in the day on 1853 Nov 9
+    # according to Astronomical Algorithms (Meeus 1998, p.191)
+    _time_first_rotation = Time('1853-11-09 21:36')
+
+    # Longitude of Earth at Carrington rotation 1 in de-tilted HCRS (so that solar north pole is Z)
+    _lon_first_rotation = \
+        get_earth(_time_first_rotation).hcrs.cartesian.transform(_sun_detilt_matrix) \
+        .represent_as(SphericalRepresentation).lon.to('deg')
+
+
+def get_sun_L0(time='now'):
+    """
+    Return the L0 angle for the Sun at a specified time, which is the Carrington longitude of the
+    Sun-disk center as seen from Earth.
+
+    Parameters
+    ----------
+    time : various
+        Time to use as `~astropy.time.Time` or in a parse_time-compatible format
+
+    Returns
+    -------
+    out : `~astropy.coordinates.Longitude`
+        The Carrington longitude
+    """
+    obstime = _astropy_time(time)
+
+    # Calculate the longitude due to the Sun's rotation relative to the stars
+    # A sidereal rotation is defined to be exactly 25.38 days
+    sidereal_lon = Longitude((obstime.jd - _time_first_rotation.jd) / 25.38 * 360*u.deg)
+
+    # Calculate the longitude of the Earth in de-tilted HCRS
+    lon_obstime = get_earth(obstime).hcrs.cartesian.transform(_sun_detilt_matrix) \
+                  .represent_as(SphericalRepresentation).lon.to('deg')
+
+    return Longitude(lon_obstime - _lon_first_rotation - sidereal_lon)
+
+
+def get_sun_P(time='now'):
+    """
+    Return the position (P) angle for the Sun at a specified time, which is the angle between
+    geocentric north and solar north as seen from Earth, measured eastward from geocentric north.
+    The range of P is +/-26.3 degrees.
+
+    Parameters
+    ----------
+    time : various
+        Time to use as `~astropy.time.Time` or in a parse_time-compatible format
+
+    Returns
+    -------
+    out : `~astropy.coordinates.Angle`
+        The position angle
+    """
+    obstime = _astropy_time(time)
+
+    geocentric = PrecessedGeocentric(equinox=obstime, obstime=obstime)
```

Comment:
This is super cool. Taken me about 5 mins to work out what it is doing. Maybe a couple of comments about `PrecessedGeocentric`?